app/controllers/annotations_controller.rb

```diff
@@ -141,4 +143,19 @@ def comment_html(medium, total_seconds, comment)
              '?time=' + total_seconds + '">' + timestamp + '</a>'
     end
 
+    def update_comments
+      for annotation in Annotation.where(medium: params[:mediumId])
+        # find comment
+        unless annotation.public_comment_id == nil
+          comment = Commontator::Comment.find_by(id: annotation.public_comment_id).body
+
+          # remove "Thymestamp: H:MM:SS" at the end of the string
+          index = comment.rindex("\nThymestamp")
+          comment = comment[0 .. index - 1]
```

Comment:
What if the "Thymestamp" part was removed by the user (by editing the comment)? What if the user inserted text after the "Thymestamp"?

Reply:
Instead, maybe search for the whole string using a regular expression that allows for a variable "H:MM:SS" part. However, we should definitely include a case when no match was found. In this case, we should probably not strip anything from the comment.

Generally, this feels like a pain point here since with the right data model, this string editing would not even be necessary, e.g. if a comment in the database had an annotation linked to it and would show the respective thymestamp in a separate HTML paragraph-element or something. However, this would include having to customize the commontator logic and UI representation which would result in another pain point :( Probably, this way is better then for right now. But again, this shows how intricate this comment-annotation relationship really is. Maybe we can improve this in the future all along a new idea: #540.

Reply:
By postponing this: do you mean completely dropping the "post as comments" feature for right now or just the part that editing the comments does not sync up with the annotation? If only the latter, it's an inconsistency we have to be aware of and users will probably notice it as a "bug". But I agree that the state of this, as of right now, is not really satisfying and this kind of string manipulation would only lead to more errors in the long run.

Reply:
I meant the latter one, i.e. you can post an annotation as a comment, but once posted the comment has nothing to do with the annotation anymore and both can be edited independently.

On the other side: If someone is not actively trying to mess up, the string manipulation won't do any harm (once, I've added regex and filter the string-not-found case).

Reply:
It could do harm depending on how you handle certain cases, e.g. when a user says

```
I really dig this new "Thymestamp 0:04:35" feature.
```

which could then become

```
I really dig this new "" feature.
```

which is semantically altering what the user was saying 😂 But of course this is an extreme edge-case (we should still have in mind though). Or what about having multiple "Thymestamp" strings in one comment, what do you do in this case?

Reply:
@fosterfarrell9 What do you think? Do you prefer

- having the "post as comment" feature right now, but without any link between comment and annotation. Maybe add a tooltip text next to the "post as comment" checkbox and make clear that editing the comment or the annotation will not affect the other side)
- having the "post as comment" feature right now with some degree of linkage between comments and annotations. In one direction (edit annotation -> update comment) full link, in other direction (edit comment -> update annotation) not at all or "hacky" with this string manipulation.
- not having the "post as comment" feature for right now and find a better solution in the future

I tend to prefer the first option where annotations can be posted as comments but editing annotations and comments is completely isolated (until a better solution, maybe with hooks into the commontator gem, is found)

Reply:
> which is semantically altering what the user was saying 😂 But of course this is an extreme edge-case (we should still have in mind though). Or what about having multiple "Thymestamp" strings in one comment, what do you do in this case?
I only catched the last occurance of "Thymestamp". So, normally: If you edit the comment, you leave the thymestamp as it is. If you edit the annotation, you won't see the thymestamp.

I'm not sure yet, which option I'd prefer. From the code side, it's pretty ugly that users have the possibility to cause unwanted behaviour. On the other side, this is not going to happen in practise, so maybe, it's better to accept this not-so-nice solution until we have something better.